assets/src/application/ui/layout/entityActionMenu/types.ts

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+
+export interface EntityActionsManager<MenuKey extends string> {
+	registerMenuItem: (key: MenuKey, render: () => React.ReactNode) => void;
+	unRegisterMenuItem: (key: MenuKey) => void;
+	getMenuItems: () => EntityMenuItems<MenuKey>;
+}
+
+export type EntityMenuItems<MenuKey extends string> = {
+	// menuKey: render()
+	[K in MenuKey]?: React.ReactNode;
+};
+
+export type MenuRegistry<EntityType extends string = '', MenuKey extends string = ''> = {
+	[K in EntityType]?: {
+		// entityId
+		[key: string]: EntityMenuItems<MenuKey>;
+	};
```

Comment:
a comment might be helpful here :smiley:

Reply:
Added a comment above there.